RationaleMCP/0031/differences.md

```diff
@@ -2,7 +2,21 @@
 This document describes differences between Flat Modelica and Modelica that aren't clear from the differences in the grammars.
 
 ## Unbalanced if-equations
-In Flat Modelica, all branches of an `if`-equation must have the same rank.  _TO BE ELABORATED…_
+In Flat Modelica, all branches of an `if`-equation must have the same number of scalarized equations,
+and if there is no `else` all branches must have zero scalarized equations.
```

Comment:
Might be easier to read if split in two sentences, with a comma after `else`.

Reply:
I see the point but on the other hand they are part of the same concept: a missing else has zero equations, and thus the other branches must have the same number of equations (i.e. zero).

Reply:
It would still be true that all branches must have the same number.  One could also just add a sentence like:
> Absence of an `else` branch is equivalent to having an empty `else` branch with "equation rank" 0.

Reply:
Not controversial. 
Decision:

- poll 
"scalar equation count" 1
"equation dimension" 1 1 1
"equation rank" 1
"equation size" 1 1 1 1 1 1 = 6 votes

Decision:
- Use term "equation size" until we have better proposal. 
- Use formulation: "Absence of an else branch is equivalent to having an empty else branch with equation size 0."
- Define "equation size" in Modelica Specification so that "local equation size" is using that definition. --> Hans
- Merge --> Hans